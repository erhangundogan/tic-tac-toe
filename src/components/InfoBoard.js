import React from 'react';

const InfoBoard = ({ isPlayersTurn, isFinished, winsCount, drawsCount, restart }) => {
  const getPlayerScoreText = (player) => {
    switch (winsCount[player]) {
      case 0: {
        return `${ player.toUpperCase() } has no winning!`;
      }
      case 1: {
        return `${ player.toUpperCase() } wins 1 time`;
      }
      default: {
        return `${ player.toUpperCase() } wins ${ winsCount[player] } times`;
      }
    }
  };

  const getStatusText = () => {
    if (isFinished) {
      return 'Game has finished!';
    }
    return `${ isPlayersTurn ? 'X' : 'O' }'s turn`;
  };

  return (
    <div className="info">
      <div>{ getStatusText() }</div>
      <div>{ getPlayerScoreText('x') }</div>
      <div>{ getPlayerScoreText('o') }</div>
      <div>{ !!drawsCount && `Draw ${drawsCount} time${drawsCount > 1 ? 's' : ''}` }</div>
      <div>{ isFinished && <button onClick={ restart }>Restart</button> }</div>
    </div>
  );
};

export default InfoBoard;
