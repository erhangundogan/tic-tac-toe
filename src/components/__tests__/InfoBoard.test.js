import React from 'react';
import { render } from '@testing-library/react';
import InfoBoard from '../InfoBoard';

describe('InfoBoard', () => {
  it('renders', () => {
    const props = {
      isPlayersTurn: true,
      isFinished: false,
      winsCount: { x: 2, o: 0 },
      drawsCount: 1,
      restart: jest.fn()
    };
    const { container } = render(
      <InfoBoard { ...props } />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
