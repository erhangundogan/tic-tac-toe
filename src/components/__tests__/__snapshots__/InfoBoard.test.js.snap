// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBoard renders 1`] = `
<div
  class="info"
>
  <div>
    X's turn
  </div>
  <div>
    X wins 2 times
  </div>
  <div>
    O has no winning!
  </div>
  <div>
    Draw 1 time
  </div>
  <div />
</div>
`;
